src/main/java/au/id/micolous/metrodroid/fragment/TripMapFragment.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.fragment.app.Fragment
+import au.id.micolous.farebot.BuildConfig
+import au.id.micolous.metrodroid.activity.TripMapActivity
+import au.id.micolous.metrodroid.transit.Trip
+import au.id.micolous.metrodroid.util.Marker
+import au.id.micolous.metrodroid.util.Preferences
+
+@RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+// We only do it only on API >= 17 and we almost don't
+// handle any user data in javascript. We use external reviewed code (Leaflet)
+@Suppress("AddJavascriptInterface", "SetJavaScriptEnabled")
+class TripMapFragment: Fragment() {
+    private var mWebView: WebView? = null
+    /**
+     * Called to instantiate the view. Creates and returns the WebView.
+     */
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        Log.d(TAG, "onCreateView")
+        mWebView?.destroy()
+        val webView = WebView(context)
+        mWebView = webView
+        webView.webChromeClient = WebChromeClient()
+        return webView
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        Log.d(TAG, "onViewCreated")
+        val webView = view as WebView
+        Log.d(TAG, "arguments=$arguments")
+        val trip = arguments?.getParcelable<Trip>(TripMapActivity.TRIP_EXTRA)
+        if (trip == null) {
+            // Probably passing around an unparcelable trip
+            Log.e(TAG, "Oops, couldn't display map, as we got a null trip!")
+            return
+        }
+
+        val settings = webView.settings
+        settings.javaScriptEnabled = true
+        settings.allowUniversalAccessFromFileURLs = true
+
+        settings.userAgentString = "${settings.userAgentString} metrodroid/${BuildConfig.VERSION_NAME}"
+
+        val tileURL = Preferences.mapTileUrl
+        val subdomains = Preferences.mapTileSubdomains
+
+        Log.d(TAG, "TilesURL: $tileURL")
+
+        Log.d(TAG, "Subdomains: $subdomains")
+
+        //int startMarkerId = R.drawable.marker_start;
+        //int endMarkerId = R.drawable.marker_end;
+
+        /* FIXME: Need icons...
+
+        if (trip.getMode() == Trip.Mode.BUS) {
+            startMarkerId = R.drawable.marker_bus_start;
+            endMarkerId   = R.drawable.marker_bus_end;
+
+        } else if (trip.getMode() == Trip.Mode.TRAIN) {
+            startMarkerId = R.drawable.marker_train_start;
+            endMarkerId   = R.drawable.marker_train_end;
+
+        } else if (trip.getMode() == Trip.Mode.TRAM) {
+            startMarkerId = R.drawable.marker_tram_start;
+            endMarkerId   = R.drawable.marker_tram_end;
+
+        } else if (trip.getMode() == Trip.Mode.METRO) {
+            startMarkerId = R.drawable.marker_metro_start;
+            endMarkerId   = R.drawable.marker_metro_end;
+
+        } else if (trip.getMode() == Trip.Mode.FERRY) {
+            startMarkerId = R.drawable.marker_ferry_start;
+            endMarkerId   = R.drawable.marker_ferry_end;
+        }
+        */
```

Comment:
We may as well remove these special icons -- we've literally had this commented out for years.  At worst we can use `git` to get it back.